src/main/java/echoserver/ClientServerConnection.java

```diff
@@ -0,0 +1,25 @@
+package echoserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.Socket;
+
+public class ClientServerConnection implements Connection{
+    public final Socket socket;
+
+    public ClientServerConnection(Socket socket) {
```

Comment:
😄 Is there a type of connection that is not a "client-server connection"?

Reply:
I guess not but can we call this class Connection (the same name as the interface)?

Reply:
Yeah, you won't be able to name `Connection` because it is already taken. I think it makes sense to keep the interface name, I am just wondering if this particular class should have a more specific name. Maybe `SocketConnection`? 🤷‍♂

Reply:
You're good with the names.  Will make the change.